### ISSUE TITLE
Extensions view: focus movement should be improved

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#116722]
1. Open the extensions viewlet and search for some extension to install "python"
2. Focus via keyboard on the Python extension item in the tree 
3. Press tab -> focus is not moved onto the Install button 🐛 

Note that this works for all other buttons but not when there is the Install button - I believe the issue comes from the fact that there is a lot action transitions happening in the background.
This is caused by my changes in the `ActionBar`, to be more precise now the action bar handles focus and outsiders should not set `tabIndex` on their actions - this is all handled by the `ActionBar`.

I have tried to fix this in Extensions, but the way the Extensions viewlet uses the ActionBar is a bit weird and I was unsuccessful.
1. It always adds all the actions to the `ActionBar` and then makes some hidden
2. It handles `tabIndex` on its own. Code pointer https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/extensions/browser/extensionsActions.ts#L728 and https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/extensions/browser/extensionsList.ts#L208

So it would be great if we could stop doing both these things. I am not sure what is the motivation behind it, there is probably some reason that I am missing.

**Comment by isidorn** [comment]
We should also aim to stop using `removeTabIndexAndUpdateFocus`. I have cleaned up all other usages of that function, currently it is only used in the extensions view.

**Comment by isidorn** [comment]
Potential workaround fix:

1. Once the action enablement is all done just call `setFocusable(true)` on the action bar. [Code pointer](https://github.com/microsoft/vscode/blob/main/src/vs/base/browser/ui/actionbar/actionbar.ts#L240) That should explicitly set the first  item to be focusable. We can just change it that it focused the first enabled item, not the first one


Here's where the `actionBar` sets the item to be focusable
https://github.com/microsoft/vscode/blob/main/src/vs/base/browser/ui/actionbar/actionbar.ts#L335

Last Friday I had a much more convoluted code, that was only making the first enabled item focusable and listening on action enablement and updating this. But could not make it work. I can revive that code if needed.

**Comment by sandy081** [comment]
@isidorn Submitted the PR - https://github.com/microsoft/vscode/pull/116964 with better and correct fix.

**Comment by isidorn** [comment]
Let's reopen this since a couple more change are need by me on the ActionBar side
